garden-backend-service/src/api/routes/modal/modal_apps.py

```diff
@@ -297,7 +224,93 @@ async def delete_modal_app(
             detail=f"Failed to delete or replace Modal App {id}. It has published children with DOIs {published_child_dois}",
         )
 
-    await db.delete(modal_app)
+
+def _validate_modal_app_metadata_helper(
+    modal_app: ModalAppCreateRequest, validate_modal_file
+):
+    _validate_modal_app_metadata(modal_app)
+    logger.info("Validated modal file metadata consistency")
+    sandbox_metadata = validate_modal_file({"file_contents": modal_app.file_contents})
+    hardware_specs = sandbox_metadata["functions"]
+    return hardware_specs
+
+
+def _generate_app_names(user: User, app_name: str) -> tuple[str, str]:
+    prefixed_app_name = f"{user.identity_id}-{app_name}"
+    full_app_name = f"{prefixed_app_name}-{str(uuid4())}"
+    return prefixed_app_name, full_app_name
+
+
+async def _handle_existing_modal_app(
+    db: AsyncSession,
+    user: User,
+    prefixed_app_name: str,
+    overwrite_existing: bool,
+    logger,
+):
+    existing_modal_app = await db.scalar(
+        select(ModalApp)
+        .where(ModalApp.user_id == user.id)
+        .filter(ModalApp.app_name.ilike(f"{prefixed_app_name}%"))
+        .order_by(ModalApp.version.desc())
+    )
+
+    logger.info("Existing Modal App", existing_modal_app=existing_modal_app)
+
+    if existing_modal_app is not None:
+        if overwrite_existing:
+            _raise_if_undeletable(existing_modal_app, user, logger)
+            logger.info(
+                "Overwriting existing modal app.", modal_app_name=prefixed_app_name
```

Comment:
I'm confused by this helper. Because its name is "_handle_existing_modal_app", I expect it to do something to the existing Modal app. But it looks like it just checks if the app name exists. I'm also confused by the "overwrite_existing" argument. When I read overwrite, I assume that something is getting deleted and something else is being written in its place. But it doesn't look like that's happening here